R/tax_expand_lat.R

```diff
@@ -0,0 +1,98 @@
+#' Generate pseudo-occurrences from latitudinal range data
+#'
+#' A function to generate pseudo-occurrences for taxa based on latitudinal
+#' ranges (e.g. the output of the 'lat' method in \code{\link{tax_range_geo}}).
+#' While the resulting pseudo-occurrences should not be treated as equivalent
+#' to actual occurrence data (e.g. like that from the Paleobiology Database),
+#' such pseudo-occurrences may be useful for  performing statistical analyses
+#' where the row representing a taxon must be replicated for each latitudinal
+#' bin through which the taxon ranges.
+#'
+#' @param taxdf \code{dataframe}. A \code{data.frame} of taxa (such as the
+#' output of the 'lat' method in \code{\link{tax_range_geo}}) with columns
+#' containing latitudinal range data (maximum and minimum latitude).
+#' Each row should represent a unique taxon. Additional columns may be included
+#' (e.g. taxon names, additional taxonomy, etc) and will be included in the
+#' returned \code{data.frame}.
+#' @param bins \code{dataframe}. A dataframe of the bins that you wish to
+#' allocate fossil occurrences to, such as that returned by
+#' \code{\link{lat_bins}}. This dataframe must contain at least the following
+#' named columns: "bin", "max" and "min".
+#' @param max_lat \code{character}. The name of the column you wish to be
```

Comment:
So I think it would be best to cut the 'max_lat' and 'min_lat' arguments out and just have columns named as 'max_lat' and 'min_lat' as required parts of the function. I have several reasons for this - firstly, you require such specifically named columns in 'bins' but don't have a seperate argument to deal with them. Second, if I were a beginner R user and was using this code, if I tried `tax_expand_lat(taxdf = test2, bins = bins, max_lat = max_lat, min_lat = min_lat)` or `tax_expand_lat(taxdf = test2, bins = bins, max_lat = test2$max_lat, min_lat = test2$min_lat)`, I would find that netiher of them worked, despite the columns being called that. I think it's also odd to have an example at the bottom that doesn't use this. Overall, I think it's somewhat unecessary, and could be confusing for people. Happy to hear alternate points of view though.

Reply:
Hi Chris, as we discussed, I think I will leave these in for now as it conforms with the rest of the package and allows great flexibility. However, I have updated the example to show explicitly how data should be input for column names!